src/lib.rs

```diff
@@ -53,29 +53,29 @@
 #![allow(clippy::too_many_arguments)]
 #![deny(rustdoc::broken_intra_doc_links)]
 #![deny(missing_docs)]
-#![no_std]
-
-#[cfg(feature = "std")]
-extern crate std;
+#![cfg_attr(not(feature = "std"), no_std)]
 
 cfg_if::cfg_if!(
 if #[cfg(feature = "alloc")] {
-    #[macro_use]
+    #[cfg_attr(not(feature = "std"), macro_use)]
```

Comment:
Could you explain what you're doing here?

Reply:
Comment added: https://github.com/dusk-network/plonk/pull/600/commits/aa1af041eda07d8b5896e97b0565f03684f353e5